cmsl1t/utils/draw.py

```diff
@@ -0,0 +1,115 @@
+                              "Fall", "FruitPunch",
+                              "Fuchsia", "GreyYellow",
+                              "GreenBrownTerrain", "GreenPink",
+                              "Island", "Lake",
+                              "LightTemperature", "LightTerrain",
+                              "Mint", "Neon",
+                              "Pastel", "Pearl",
+                              "Pigeon", "Plum",
+                              "RedBlue", "Rose",
+                              "Rust", "SandyTerrain",
+                              "Sienna", "Solar",
+                              "SouthWest", "StarryNight",
+                              "Sunset", "TemperatureMap",
+                              "Thermometer", "Valentine",
+                              "VisibleSpectrum", "WaterMelon",
+                              "Cool", "Copper",
+                              "GistEarth", "Viridis"])
+
+
+def root_palette(value, max, min=0):
+    colour_index = float(value - min) / float(max - min) * gStyle.GetNumberOfColors()
+    return gStyle.GetColorPalette(int(colour_index))
+
+
+def __clean(hists):
+    cleaned_hists = []
+    for hist in hists:
+        if isinstance(hist, Efficiency):
+            hist = hist.graph
+        cleaned_hists.append(hist)
+    return cleaned_hists
+
+
+def __apply_colour_map(hists, colourmap, colour_values, change_colour):
+    # Clean change_colour
+    change_colour = [c.lower() for c in change_colour]
+
+    with preserve_current_style():
+        # Resolve the requested pallette if it's not a function
+        if isinstance(colourmap, str):
+            if colourmap in __known_root_pallettes:
+                gStyle.SetPalette(getattr(ROOT, "k" + colourmap))
+                colourmap = root_palette
+            else:
+                raise RuntimeError("Unknown palette requested: " + colourmap)
+
+        # Set the colour of each hist
+        max = len(hists)
+        for value, hist in enumerate(hists):
+            if colour_values:
+                value, max = colour_values(value)
+            colour = colourmap(value, max)
+            if "line" in change_colour:
+                hist.linecolor = colour
+            if "marker" in change_colour:
+                hist.markercolor = colour
+
+
+def __prepare_canvas():
+    style = gStyle
+    style.SetOptStat(0)
+
+    canvas = Canvas()
+    canvas.SetGridx(True)
+    canvas.SetGridy(True)
+    return canvas, style
+
+
+def draw(hists, colourmap="RainBow", colour_values=None,
+         change_colour=("line", "marker")):
+    canvas, style = __prepare_canvas()
+    hists = __clean(hists)
+    __apply_colour_map(hists, colourmap, colour_values, change_colour)
+    r_draw(hists, canvas)
+    return canvas
+
+
+def label_canvas(sample_title=None, run=None, isData=False):
+    latex = TLatex()
+    latex.SetNDC()
+    latex.SetTextFont(42)
+
+    cms = "#bf{CMS} #it{Preliminary}"
```

Comment:
Not sure, what's the benefit?  Think it makes the code less clear so I'd prefer to leave it as is.